### PR TITLE
Require explicit CNN model path for issue 185

### LINKED ISSUE #185: [Task] Remove implicit experiment-directory fallback from pipeline CNN model resolution
## Background

During #45 / PR #184, CNN-related scripts under `tools/` and `experiments/` were inventoried and one-off scripts were removed or moved.

The cleanup confirmed that the current production pipeline no longer imports `tools/cnn_classifier/score_candidates_batch.py`; `src/pipeline/detection/orchestrator.py` calls `src.pipeline.steps.cnn_scoring.run_cnn_scoring_batch` directly.

However, `src/pipeline/steps/cnn_scoring.py` still has an implicit fallback in `_resolve_model_path()`:

```python
candidates = list(Path("experiments/cnn_classifier").rglob("best_model.pth"))
```

This is not a direct dependency on a tool script, but it means pipeline behavior can depend on experiment-directory contents when `detection.cnn_model_path` is wrong or missing.

## Goal

Make pipeline CNN model resolution explicit and independent from `experiments/` layout.

## Scope

- Remove or replace the fallback search under `experiments/cnn_classifier/**/best_model.pth`.
- Ensure `detection.cnn_model_path` is required and failure is explicit when the configured model is missing.
- Keep production pipeline behavior deterministic.
- Update docs/config comments if they currently imply that experiment-directory fallback is supported.

## Non-goals

- Do not reorganize CNN training scripts again.
- Do not change CNN model architecture, thresholds, or evaluation metrics.
- Do not run full Stage E/full-pipeline validation unless the code change requires it.

## Suggested validation

```bash
git diff --check
PYTHONPATH=. python3 -m py_compile src/pipeline/steps/cnn_scoring.py src/pipeline/detection/orchestrator.py
make test-fast
```

Add or update a focused unit test if existing tests cover missing-model error handling.

## Related

- #45
- PR #184

### PR DESCRIPTION
**Pull request by M763468**
Summary:

- Remove the fallback search under `experiments/cnn_classifier/**/best_model.pth` from `src/pipeline/steps/cnn_scoring.py`.
- Require `detection.cnn_model_path` to point to an explicitly configured model file.
- Raise explicit errors for missing paths, directory paths, and missing `detection.cnn_model_path` configuration.
- Normalize string model paths through `Path(...)`.
- Add focused tests for model path resolution behavior, including `Path`, `str`, directory, missing-file, and `None` cases.

Validation:

Maintainer confirmed the following local checks passed after fast-forwarding to the latest PR head:

```bash
git diff --check
PYTHONPYCACHEPREFIX=/tmp/pycache-issue185 \
PYTHONPATH=. .venv_pdf/bin/python -m py_compile \
  src/pipeline/steps/cnn_scoring.py \
  tests/test_cnn_scoring_model_path.py
PYTHONPATH=. .venv_pdf/bin/python -m pytest \
  tests/test_cnn_scoring_model_path.py
```

Result:

```text
collected 5 items
tests/test_cnn_scoring_model_path.py ..... [100%]
5 passed
```

Closes #185.

### CODE REVIEW

File: src/pipeline/steps/cnn_scoring.py [hunk 1]
```diff
@@ -38,12 +38,19 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
...
+        raise FileNotFoundError(
+            "CNN model path is not a file. "
+            "Configure detection.cnn_model_path to an existing model file. "
+            f"Configured path: {model_path}"
+        )
+    raise FileNotFoundError(
+        "CNN model file not found. "
+        "Configure detection.cnn_model_path to an existing model file. "
+        f"Configured path: {model_path}"
+    )
```
[the remark this answers is not in the document]
Comment (M763468) [reply src/pipeline/steps/cnn_scoring.py]: 対応しました。

- `_resolve_model_path` now accepts `Path | str | None`.
- `None` raises an explicit `ValueError` for missing `detection.cnn_model_path` configuration.
- String inputs are normalized through `Path(...)` before file/directory checks.
- Added focused tests for string path and `None` cases.
